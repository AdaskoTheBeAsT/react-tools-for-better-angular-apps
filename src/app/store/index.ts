export * from './actions';
export { getCount, getRouter, State } from './reducers';

export { StoreModule } from './store.module';
