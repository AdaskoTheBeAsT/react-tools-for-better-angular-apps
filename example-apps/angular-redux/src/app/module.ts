import { BrowserModule } from '@angular/platform-browser'
import { NgModule } from '@angular/core'
import { FormsModule } from '@angular/forms'
import { RouterModule } from '@angular/router'
import { HttpClientModule } from '@angular/common/http'
import { NgReduxModule } from '@angular-redux/store'
import { NgReduxRouterModule } from '@angular-redux/router'

// This app's ngModules
import { StoreModule } from './store/module'
import { AnimalModule } from './animals/module'
import { ElephantModule } from './elephants/module'
import { LionModule } from './lions/module'
import { FeedbackModule } from './feedback/module'

// Top-level app component constructs.
import { appRoutes } from './routes'
import { AppComponent } from './component'

@NgModule({
  declarations: [AppComponent],
  imports: [
    RouterModule.forRoot(appRoutes),
    BrowserModule,
    FormsModule,
    HttpClientModule,
    NgReduxModule,
    NgReduxRouterModule,
    AnimalModule,
    ElephantModule,
    LionModule,
    FeedbackModule,
    StoreModule,
  ],
  bootstrap: [AppComponent],
})
export class AppModule {}
