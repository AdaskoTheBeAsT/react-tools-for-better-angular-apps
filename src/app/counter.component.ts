import { Component, OnInit, ChangeDetectionStrategy } from '@angular/core';
import { select, NgRedux } from '@angular-redux/store';
import { Observable } from 'rxjs/Observable';

import * as fromRoot from './store';

@Component({
  selector: 'app-counter',
  template: `
    <button (click)="increment()">inc</button>
    <button (click)="decrement()">dec</button>
    <p>Clicked {{ count$ | async }} times</p>
  `,
  styles: [],
  changeDetection: ChangeDetectionStrategy.OnPush,
})
export class CounterComponent implements OnInit {
  constructor(private ngRedux: NgRedux<fromRoot.State>) {}
  count$ = this.ngRedux.select(fromRoot.getCount);

  increment() {
    this.ngRedux.dispatch(new fromRoot.IncAction());
  }
  decrement() {
    this.ngRedux.dispatch(new fromRoot.DecAction());
  }

  ngOnInit() {}
}
