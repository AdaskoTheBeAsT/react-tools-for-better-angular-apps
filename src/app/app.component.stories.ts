import { storiesOf, moduleMetadata } from '@storybook/angular';

import { AppComponent } from './app.component';
import { AppModule } from './app.module';

storiesOf('App Component', module).add('root app component', () => ({
  // @FIXME: aliasing doesn't work here !
  moduleMetadata: {
    // imports: [AppModule],
  },
  // template: `<app-root></app-root>`,
  template: `@FIXME Martin !`,
  props: {},
}));
