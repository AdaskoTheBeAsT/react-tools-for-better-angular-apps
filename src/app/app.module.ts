import { BrowserModule } from '@angular/platform-browser';
import { NgModule } from '@angular/core';
import { Routes, RouterModule } from '@angular/router';

import { DemoModule } from './demo/demo.module';

import { AppComponent } from './app.component';

@NgModule({
  declarations: [AppComponent],
  imports: [BrowserModule, DemoModule, RouterModule.forRoot([])],
  providers: [],
  bootstrap: [AppComponent],
})
export class AppModule {}
