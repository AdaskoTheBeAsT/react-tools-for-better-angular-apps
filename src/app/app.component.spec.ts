import { TestBed, async } from '@angular/core/testing';
import { RouterTestingModule } from '@angular/router/testing';

import { AppComponent } from './app.component';
describe('AppComponent', () => {
  beforeEach(
    async(() => {
      TestBed.configureTestingModule({
        imports: [RouterTestingModule],
        declarations: [AppComponent],
      }).compileComponents();
    })
  );
  it(
    'should create the app',
    async(() => {
      const fixture = TestBed.createComponent(AppComponent);
      const app = fixture.debugElement.componentInstance;

      expect(app).toBeTruthy();
      expect(fixture).toMatchSnapshot();
    })
  );
  it(
    `should have no properties`,
    async(() => {
      const fixture = TestBed.createComponent(AppComponent);
      const app = fixture.debugElement.componentInstance;
      const actual = Object.keys(app);

      expect(actual).toEqual([]);
    })
  );
  it(
    'should render title in a h1 tag',
    async(() => {
      const fixture = TestBed.createComponent(AppComponent);
      fixture.detectChanges();
      const compiled = fixture.debugElement.nativeElement;

      expect(compiled.querySelector('img').alt).toContain('Ultimate Pizza');
      expect(fixture).toMatchSnapshot();
    })
  );
});
