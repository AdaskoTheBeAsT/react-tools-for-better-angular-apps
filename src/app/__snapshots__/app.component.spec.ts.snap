// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AppComponent should create the app 1`] = `
<app-root>
  
    <div
    class="app"
  >
    
      
    <div
      class="app__header"
    >
      
        
      <img
        alt="Ultimate Pizza"
        class="app__logo"
        src="/assets/img/logo.svg"
      />
      
      
    </div>
    
      
    <div
      class="app__content"
    >
      
        
      <div
        class="app__nav"
      >
        
          
        <a
          routerlink="products"
          routerlinkactive="active"
        >
          Products
        </a>
        
        
      </div>
      
        
      <div
        class="app__container"
      >
        
          
        <router-outlet />
        
        
      </div>
      
        
      <div
        class="app__footer"
      >
        
          
        <p>
          © Ultimate Pizza Inc.
        </p>
        
        
      </div>
      
      
    </div>
    
    
  </div>
    
</app-root>
`;

exports[`AppComponent should render title in a h1 tag 1`] = `
<app-root>
  
    <div
    class="app"
  >
    
      
    <div
      class="app__header"
    >
      
        
      <img
        alt="Ultimate Pizza"
        class="app__logo"
        src="/assets/img/logo.svg"
      />
      
      
    </div>
    
      
    <div
      class="app__content"
    >
      
        
      <div
        class="app__nav"
      >
        
          
        <a
          href="/products"
          ng-reflect-router-link="products"
          ng-reflect-router-link-active="active"
          routerlink="products"
          routerlinkactive="active"
        >
          Products
        </a>
        
        
      </div>
      
        
      <div
        class="app__container"
      >
        
          
        <router-outlet />
        
        
      </div>
      
        
      <div
        class="app__footer"
      >
        
          
        <p>
          © Ultimate Pizza Inc.
        </p>
        
        
      </div>
      
      
    </div>
    
    
  </div>
    
</app-root>
`;
